### DISCUSSION TITLE
errors on :checkhealth

### DISCUSSION THREAD

**Discussion by loginAsOne** [LazyVim/LazyVim#2477] [Q&A] [answered]
I am currently getting errors on running :checkhealth

first one is 

- ERROR Parser "jsdoc" failed to load (path: /home/login/.local/share/nvim/lazy/nvim-treesitter/parser/jsdoc.so): Failed to load parser for language 'jsdoc': uv_dlopen: /home/login/.local/share/nvim/lazy/nvim-treesitter/parser/jsdoc.so: undefined symbol: tree_sitter_jsdoc_external_scanner_create

and second one is

which-key: require("which-key.health").check()

- ERROR Failed to run healthcheck for "which-key" plugin. Exception:
  function health#check, line 25
  Vim(eval):E5108: Error executing lua ...al/share/nvim/lazy/which-key.nvim/lua/which-key/keys.lua:405: Invalid buffer id: 2
  stack traceback:
  [C]: in function 'nvim_buf_get_keymap'
  ...al/share/nvim/lazy/which-key.nvim/lua/which-key/keys.lua:405: in function 'update_keymaps'
  .../share/nvim/lazy/which-key.nvim/lua/which-key/health.lua:15: in function 'check'
  [string "luaeval()"]:1: in main chunk

does anyone know how to resolve these issue? I am new to lazyvim

**Comment by dil-mnasir** [comment]
I am also getting the same after typescript treesitter update. 
@amaanq

**Reply by amaanq** [reply]
a jsdoc error is unrelated to typescript I would say. also, you need to ensure your parsers are up-to-date when updating nvim-treesitter

**Reply by dil-mnasir** [reply]
This is not the only error. When I open any typescript files it throws this error: 
![image](https://github.com/LazyVim/LazyVim/assets/93813665/9e5f55e9-963f-46b1-b4c7-e5c0dbdc5663)

P.S All my parsers are up-to-date btw.

Is there any way I can downgrade and check the behavior. I am sure I wasn't getting this error yesterday.

**Reply by amaanq** [reply]
ah I think the text objects queries need to be updated too, thanks for the reminder I'll PR that in a min

**Reply by amaanq** [reply]
try now @mohnasir-cisco (update nvim-ts-textobjects)

**Reply by dil-mnasir** [reply]
Thanks, its working fine now.

**Reply by loginAsOne** [reply]
@amaanq for jsdoc parser, how can I update this?

**Reply by loginAsOne** [reply, marked as answer]
Updated with jsdoc using npm update -g jsdoc and updated lazyvim and synced. It is working now
